### DISCUSSION TITLE
Sort wave table filter items

### DISCUSSION THREAD

**Discussion by shamilh2o** [h2oai/wave#2115] [Q&A] [answered]
Het team,

The filter feature in wave table columns is really cool. I want to know how can we sort entires alphabetically which are listing in the filter dropdown (shown below).
![image](https://github.com/h2oai/wave/assets/101506482/062eef83-093b-4755-b36a-fee428e98a68)

I just passed the sorted list of strings and assigned to the filters attribute in the table column (filters=['Cat', 'Dog', 'Lion', 'Zeebra'] but it didn't give any response for me. 

I would like to know whether it's possible and if so how. 
Thanks!
cc: @mturoci

**Comment by mturoci** [comment]
> I just passed the sorted list of strings and assigned to the filters attribute in the table column (filters=['Cat', 'Dog', 'Lion', 'Zeebra'] but it didn't give any response for me.

Please provide a MRE and Wave version.

**Reply by shamilh2o** [reply]
Sorry I missed to pass it. 
My actual setup is in wave 0.25.3 which faced the issue.
The code below is the one in "Wave Tour 0.25.2" app, which is slightly modified the notification item and adding "filters" attribute.
Even I manually passed the list by sorted, the filter dropdown does not pick it. Is that that normal behaviour? Can we sort it down?
```

import random
from faker import Faker
from h2o_wave import main, app, Q, ui

fake = Faker()

_id = 0


class Issue:
    def __init__(self, text: str, status: str, progress: float, icon: str, notifications: str):
        global _id
        _id += 1
        self.id = f'I{_id}'
        self.text = text
        self.status = status
        self.views = 0
        self.progress = progress
        self.icon = icon
        self.notifications = notifications


# Create some issues
issues = [
    Issue(
        text=fake.sentence(),
        status=('Closed' if i % 2 == 0 else 'Open'),
        progress=random.random(),
        icon=('BoxCheckmarkSolid' if random.random() > 0.5 else 'BoxMultiplySolid'),
        notifications=('Cat' if random.random() > 0.5 else 'Dog' if random.random() > 0.5 else 'Lion' if random.random() > 0.5 else "Zeebra")) for i in range(100)
]

# Create columns for our issue table.
columns = [
    ui.table_column(name='text', label='Issue'),
    ui.table_column(name='status', label='Status', filterable=True),
    ui.table_column(name='notifications', label='Notifications', filterable=True, filters=['Cat', 'Dog', 'Lion', 'Zeebra']),
    ui.table_column(name='done', label='Done', cell_type=ui.icon_table_cell_type()),
    ui.table_column(name='views', label='Views'),
    ui.table_column(name='progress', label='Progress', cell_type=ui.progress_table_cell_type()),
]


@app('/demo')
async def serve(q: Q):
    q.page['form'] = ui.form_card(box='1 1 -1 7', items=[
        ui.table(
            name='issues',
            columns=columns,
            rows=[ui.table_row(
                name=issue.id,
                cells=[issue.text, issue.status, issue.notifications, issue.icon, str(issue.views), str(issue.progress)]) for issue in issues]
        )
    ])
    await q.page.save()
    

```
![image](https://github.com/h2oai/wave/assets/101506482/78d02b47-3454-4167-89cb-346c1bb48cac)

Just checked whether it sorts after sorting the table column values (by adding sortable attribute),
`ui.table_column(name='notifications', label='Notifications', filterable=True, sortable=True, filters=['Cat', 'Dog', 'Lion', 'Zeebra']),`

But still it takes the initial order. 
![image](https://github.com/h2oai/wave/assets/101506482/dd98f462-437b-42ca-ae82-50ac17501624)

**Reply by mturoci** [reply, marked as answer]
TLDR: Upgrade to `0.26+`.

> Even I manually passed the list by sorted, the filter dropdown does not pick it. Is that that normal behaviour? Can we sort it down?

In versions `<0.26` yes. The explicit filters were supposed to be used on paginated tables only. From `0.26.0` this constraint was removed and filters can be set regardless of whether the table is paginated or client-side.
